### PR TITLE
docs: add release notes for 0.18.4

### PR DESCRIPTION
**Pull request by eecsliu**
Add release notes for 0.18.4.

### CODE REVIEW

**Review by vishnu2kmohan — APPROVED** [review]
Thanks @eecsliu!

File: docs/release-notes.txt [hunk 1]
```diff
@@ -8,6 +8,34 @@
...
+
+**Release Date:** July 14, 2022
+
+**New Features**
+
+-  Configuration: Add support for ``task_container_defaults.environment_variables`` in master
+   config. Users can now specify a list of environment variables that will be set in all task
+   containers.
+
+-  Web: Most user settings and preferences, like filters, are now persisted to the db. Users will
```
Comment (dzhu) [inline docs/release-notes.txt:22]:
```suggestion
-  Web: Most user settings and preferences, like filters, are now persisted to the database. Users will
```

File: docs/release-notes.txt [hunk 2]
```diff
@@ -8,6 +8,34 @@
...
+   now be able to retain their settings across devices.
+
+**Bug Fixes**
+
+-  Since 0.17.7, ``det experiment download-model-def $ID`` has been saving the downloaded tarballs
+   as just ``$ID``. This release corrects that behavior and names them
+   ``experiment_$ID_model_def.tgz`` instead.
+
+-  Kubernetes: fix a bug where following the link to live tensorboards would redirect to the
```
Comment (dzhu) [inline docs/release-notes.txt:31]:
```suggestion
-  Kubernetes: Fix a bug where following the link to live TensorBoards would redirect to the
```

File: docs/release-notes.txt [hunk 3]
```diff
@@ -8,6 +8,34 @@
...
+   ``Uncategorized`` page.
+
+**Improvements**
+
+-  Adds the resource pool field when listing experiments or commands in k8s, where it was previously
```
Comment (dzhu) [inline docs/release-notes.txt:36]:
```suggestion
-  Add the resource pool field when listing experiments or commands in Kubernetes, where it was previously
```

**Pull request merged**